commands/login_test.go

```diff
@@ -415,11 +415,41 @@ func TestCreateSSHCert(t *testing.T) {
 }
 
 func TestIdentityString(t *testing.T) {
-	pkt, _, _ := Mocks(t, ECDSA)
-	idString, err := IdentityString(*pkt)
-	require.NoError(t, err)
-	expIdString := "Email, sub, issuer, audience: \narthur.aardvark@example.com me https://accounts.example.com test_client_id"
-	require.Equal(t, expIdString, idString)
+	t.Run("with email claim", func(t *testing.T) {
+		pkt, _, _ := Mocks(t, ECDSA)
+		idString, err := IdentityString(*pkt)
+		require.NoError(t, err)
+		expIdString := "Email, sub, issuer, audience: \narthur.aardvark@example.com me https://accounts.example.com test_client_id"
+		require.Equal(t, expIdString, idString)
+	})
+
+	t.Run("without email claim", func(t *testing.T) {
+		// Create a mock without email claim
+		alg := jwa.ES256
+		signer, err := util.GenKeyPair(alg)
+		require.NoError(t, err)
+
+		providerOpts := providers.DefaultMockProviderOpts()
+		op, _, idtTemplate, err := providers.NewMockProvider(providerOpts)
+		require.NoError(t, err)
+
+		// Explicitly set ExtraClaims to empty to ensure no email claim is present in the ID token
+		idtTemplate.ExtraClaims = map[string]any{}
```

Comment:
Instead of all this duplicated code, try modifying the Mocks function to take an optional argument with the ExtraClaims the test wants to specify.